java/analyze-hotspots/src/main/java/com/esri/arcgisruntime/sample/analyzehotspots/MainActivity.java

```diff
@@ -16,39 +16,242 @@
+            findProgressDialogFragment().dismiss();
+          }
+        });
+      } catch (InterruptedException | ExecutionException e) {
+        e.printStackTrace();
+      }
+    });
+  }
+
+  private void addLayerToMapView(final Layer layer, final MapView mapView) {
+    // add the new layer to the map
+    mapView.getMap().getOperationalLayers().add(layer);
+
+    // Add a listener to set the viewpoint of the map view when it is finished loading only if the layer has not already
+    // finished loading
+    if (layer.getLoadStatus() != LoadStatus.LOADED) {
+      layer.addDoneLoadingListener(() -> {
+        // set the map viewpoint to the MapImageLayer, once loaded
+        mMapView.setViewpointGeometryAsync(layer.getFullExtent());
+      });
+    }
+  }
+
+  void cancelGeoProcessingJob() {
+    mCancelled = true;
+    if (mGeoprocessingJob != null) {
+      mGeoprocessingJob.cancel();
+    }
+  }
+
+  private DateRangeDialogFragment findDateRangeDialogFragment() {
+    return (DateRangeDialogFragment) getSupportFragmentManager()
+        .findFragmentByTag(DateRangeDialogFragment.class.getSimpleName());
+  }
+
+  private ProgressDialogFragment findProgressDialogFragment() {
+    return (ProgressDialogFragment) getSupportFragmentManager()
+        .findFragmentByTag(ProgressDialogFragment.class.getSimpleName());
+  }
+
+  private void showDateRangeDialog() {
+    if (findDateRangeDialogFragment() == null) {
+      DateRangeDialogFragment dateRangeDialogFragment = DateRangeDialogFragment.newInstance(
+          getString(R.string.date_range_dialog_title),
+          getString(R.string.date_range_dialog_submit_button_text)
+      );
+      dateRangeDialogFragment.show(getSupportFragmentManager(), DateRangeDialogFragment.class.getSimpleName());
+    }
+  }
+
+  private void showProgressDialogFragment() {
+    if (findProgressDialogFragment() == null) {
+      ProgressDialogFragment progressDialogFragment = ProgressDialogFragment.newInstance(
+          getString(R.string.progress_dialog_fragment_title),
+          getString(R.string.progress_dialog_fragment_cancel_text)
+      );
+      progressDialogFragment.show(getSupportFragmentManager(), ProgressDialogFragment.class.getSimpleName());
+    }
   }
 
   @Override public void onProgressDialogCancel() {
-    findAnalyzeHotspotsFragment().cancelGeoProcessingJob();
+    cancelGeoProcessingJob();
   }
 
-  private AnalyzeHotspotsFragment findAnalyzeHotspotsFragment() {
-    AnalyzeHotspotsFragment fragment = (AnalyzeHotspotsFragment) getSupportFragmentManager()
-        .findFragmentByTag(AnalyzeHotspotsFragment.class.getSimpleName());
-    if (fragment != null) {
-      return fragment;
-    } else {
-      return AnalyzeHotspotsFragment.newInstance();
+  @Override
+  public void onResume() {
+    super.onResume();
+    mMapView.resume();
+
+    // If the layer has already been created, add it to the map view
+    if (mHotspotMapImageLayer != null) {
```

Comment:
do we still need these calls if we're not doing the geoprocessing in a fragment and/or handling config changes in manifest?

Reply:
The idea behind adding the config changes was to allow us to persist these over rotation, so these fields are still being persistsed.

Reply:
* we check if it's null to check if it's been created or not.

Reply:
But aren't we only ever hitting onResume once now, when the app launches?

Reply:
Right enough, I was under the impression the lifecycle was still happening. Will remove.